### PR TITLE
Remove SQSReader from worker

### PR DESCRIPTION
**Pull request by kenoir**
### What is this PR trying to achieve?

Simplify the `MessageWorker` by pushing the `SQSReader` inside the `MessageReader` class.

### Who is this change for?

🍔 

### Have the following been considered/are they needed?

- [ ] Deployed new versions

### CODE REVIEW

File: catalogue_pipeline/id_minter/src/test/scala/uk/ac/wellcome/platform/idminter/steps/IdEmbedderTests.scala [hunk 1]
```diff
@@ -23,7 +23,7 @@ class IdEmbedderTests
     with BeforeAndAfterEach
     with MockitoSugar
     with JsonTestUtil
-    with PatienceConfiguration {
+    with ExtendedPatience {
```
Comment (alexwlchan) [inline catalogue_pipeline/id_minter/src/test/scala/uk/ac/wellcome/platform/idminter/steps/IdEmbedderTests.scala]: If you're replacing PatienceConfiguration with ExtendedPatience, are you going to remove the patienceConfig below?

File: sbt_common/messaging/src/test/scala/uk/ac/wellcome/messaging/test/fixtures/Messaging.scala [hunk 2]
```diff
@@ -206,22 +218,52 @@ trait Messaging
...
-              }
-          }
+            }
         }
+
       }
     }
   }
 
+  def assertQueueEmpty(queue: Queue) = {
```
Comment (alexwlchan) [inline sbt_common/messaging/src/test/scala/uk/ac/wellcome/messaging/test/fixtures/Messaging.scala]: Would these be more usefully defined on the SQS fixture, so they're available to other tests?

**Comment by alexwlchan** [comment]
This PR also seems to be splitting MessageConfig into MessageReaderConfig and MessageWriterConfig. Could we get those changes as a separate PR?

File: catalogue_pipeline/id_minter/src/test/scala/uk/ac/wellcome/platform/idminter/steps/IdEmbedderTests.scala (hunk 1 above)
Comment (alicefuzier) [reply catalogue_pipeline/id_minter/src/test/scala/uk/ac/wellcome/platform/idminter/steps/IdEmbedderTests.scala]: oh thanks

File: sbt_common/messaging/src/test/scala/uk/ac/wellcome/messaging/test/fixtures/Messaging.scala (hunk 2 above)
Comment (alicefuzier) [reply sbt_common/messaging/src/test/scala/uk/ac/wellcome/messaging/test/fixtures/Messaging.scala]: yep, will move

**Comment by alicefuzier** [comment]
Addressed review comments. As for splitting the PR, the commit history is in a state where it would be a bit of a hassle to split. Also, the reason for splitting the config class came from the changes to `MessageReader`, so I don't think they are totally independent changes

**Pull request merged**